### PR TITLE
Add D-Bus tests for the two filter variants

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: None
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

File: tests/tests/test_monitor_client.py [hunk 1]
```diff
@@ -657,3 +663,73 @@ def test_dbus_subsystem(self, monitor_commercial_setup_no_client):
...
+        expected_from = "alert@mender.io"
+        dbus_name = "test"
+        user_name = "bugs.bunny@acme.org"
+        devid, _, _, mender_device = self.prepare_env(
+            monitor_commercial_setup_no_client, user_name
+        )
+        logger.info("test_dbus_subsystem: env ready.")
+
+        logger.info(
+            "test_dbus_subsystem: email alert on dbus signal pattern match scenario."
```
Comment (merlin-northern) [inline tests/tests/test_monitor_client.py:680]: nit: "test_dbus_subsystem" -> "test_dbus_pattern_match"

File: tests/tests/test_monitor_client.py [hunk 2]
```diff
@@ -657,3 +663,73 @@ def test_dbus_subsystem(self, monitor_commercial_setup_no_client):
...
+        )
+        prepare_dbus_monitoring(mender_device, dbus_name, log_pattern="mender")
```
Comment (merlin-northern) [inline tests/tests/test_monitor_client.py:682]: another nit: technically it is not "log_pattern" but `dbus_pattern="mender"` right?

File: tests/tests/test_monitor_client.py [hunk 3]
```diff
@@ -144,6 +146,10 @@ def prepare_dbus_monitoring(mender_device, dbus_name):
         dbus_check_file = os.path.join(tmpdir, "dbus_%s.sh" % dbus_name)
         f = open(dbus_check_file, "w")
         f.write("DBUS_NAME=%s\n" % dbus_name)
+        if log_pattern:
+            f.write("DBUS_PATTERN=%s\n" % log_pattern)
+        if bus_pattern:
+            f.write("DBUS_WATCH_BATTERN=%s\n" % bus_pattern)
```
Comment (merlin-northern) [inline tests/tests/test_monitor_client.py:152]: berhabs "DBUS_WATCH_BATTERN" -> "DBUS_WATCH_PATTERN"

File: tests/tests/test_monitor_client.py [hunk 4]
```diff
@@ -657,3 +663,73 @@ def test_dbus_subsystem(self, monitor_commercial_setup_no_client):
...
+        user_name = "bugs.bunny@acme.org"
+        devid, _, _, mender_device = self.prepare_env(
+            monitor_commercial_setup_no_client, user_name
+        )
+        logger.info("test_dbus_subsystem: env ready.")
+
+        logger.info(
+            "test_dbus_subsystem: email alert on single dbus filter signal scenario."
+        )
+        prepare_dbus_monitoring(mender_device, dbus_name, log_pattern="mender")
```
Comment (merlin-northern) [inline tests/tests/test_monitor_client.py:717]: in both places you are passing log_pattern, so, where are we testing WATCH_PATTERN

**Review by merlin-northern — COMMENTED** [review]
general comment: I am missing a place where we actually trigger the signals?

File: tests/tests/test_monitor_client.py (hunk 2 above)
Comment (oleorhagen) [reply tests/tests/test_monitor_client.py:682]: Right, good catch. 
Thx

File: tests/tests/test_monitor_client.py (hunk 3 above)
Comment (oleorhagen) [reply tests/tests/test_monitor_client.py:152]: haha, berhaps :see_no_evil:

**Comment by merlin-northern** [comment]
> general comment: I am missing a place where we actually trigger the signals?

treat as "nice to have" -- discussed over slack

**Pull request merged**